datastore/gcloud/aio/datastore/value.py

```diff
@@ -35,6 +36,8 @@ def from_repr(cls, data: Dict[str, Any]) -> 'Value':
             if json_key in data:
                 if json_key == 'nullValue':
                     value = None
+                elif value_type == Array:
```

Comment:
I like what you did in your other PR for merging the `value_type.from_repr` cases -- no need to change this, since it'll annoy us with merge conflicts, this comment is just a note to myself to merge these too approaches when I merge both your PRs.